core/peer/config.go

```diff
@@ -37,9 +37,13 @@ import (
 // ExternalBuilder represents the configuration structure of
 // a chaincode external builder
 type ExternalBuilder struct {
-	EnvironmentWhitelist []string `yaml:"environmentWhitelist"`
-	Name                 string   `yaml:"name"`
-	Path                 string   `yaml:"path"`
+	// TODO: Remove EnvironmentList in 3.0
+	// Deprecated: EnvironmentList is retained for backwards compatibility.
+	// New deployments should use the new EnvPropagationList field
+	EnvironmentList    []string `yaml:"environmentWhitelist"`
+	EnvPropagationList []string `yaml:"envPropagationList"`
```

Comment:
No need for a list suffix on a list. `env` as an abbreviation in code is okay but I believe using the full name in the config key is more clear.

Reply:
I amended it, the only issue I had with the proposed name in Jira `propagateEnvironment` is that it implies a boolean rather than an array. So I left the tag as having a the `List` suffix on it as `propagateEnvironmentList`